src/service/back-in-stock.service.ts

```diff
@@ -47,11 +54,54 @@ export class BackInStockService {
         private productVariantService: ProductVariantService,
         private translatorService: TranslatorService,
         private eventBus: EventBus,
-    ) {}
+        @Inject(PLUGIN_INIT_OPTIONS) private options: BackInStockOptions
+    ) { }
 
-    async findOne(ctx: RequestContext, id: ID): Promise<BackInStock | undefined> {
-        return this.connection.getRepository(ctx, BackInStock).findOne(id, {
-            relations: this.relations,
+
+    onApplicationBootstrap() {
+        // Listen for stock movements and update subscriptions
+        this.eventBus.ofType(StockMovementEvent).subscribe(async event => {
+            // Refetch variants, because variants in event does not have all properties fetched from DB
+            const variants = await this.productVariantService.findByIds(event.ctx, event.stockMovements.map(sm => sm.productVariant.id));
```

Comment:
`event.stockMovements[0].productVariant` only has a partial variant: `{id: 1}`, so we refetch the entire variant to be able to resolve `saleableStockLevel`